src/pages/member/UserHome.js

```diff
@@ -1,15 +1,175 @@
+        fontSize: '36px',
+        paddingLeft: '76px',
+        paddingTop: '45px'
+    },
+    reward: {
+        fontStyle: 'normal',
+        fontWeight: 'normal',
+        fontSize: '22px',
+        paddingLeft: '76px',
+        paddingTop: '15px',
+    },
+    house: {
+        position: 'absolute',
+        left: '685px',
+        top: '110px'
+    }
+}
+
+const useStyles = makeStyles({
+    root: {
+        width: '719px',
+        marginTop: '27px'
+    },
+})
+
+function UserHome(props) {
+    const [featuredEvent, setFeaturedEvent] = useState()
+    const [nextEvent, setNextEvent] = useState()
+    const getFeaturedEvent = () => {
+        if (props.events && props.events.length) {
+            setFeaturedEvent(props.events[Math.floor(Math.random() * (props.events.length - 1))]);
+        }
+    }
+
+    /**
+     * gets the next event that the user is registered for
+     * verifies that the event is after the current time
+     * sets next event to 'None Registered!' if no events found
+     */
+    const getNextEvent = async () => {
+        const params = new URLSearchParams({
+            id: props.user.id
+        })
+        await fetchBackend(`/registrations?${params}`, 'GET')
+            .then(async response => {
+                if (response && response.size > 0) {
+                    //iterate over events - the first one that is found in registrations is the closest event assuming that events are already sorted by date
+                    let found = false;
+                    if (props.events) {
+                        props.events.forEach(event => {
+                            if (!found) {
+                                const index = response.data.findIndex(registration => registration.eventID === event.id)
+                                if (index !== -1) {
+                                    found = true;
+                                    // if the event has not passed yet
+                                    if (new Date(event.startDate).getTime() > new Date().getTime()) {
+                                        setNextEvent(event)
+                                    } else {
+                                        setNextEvent({
+                                            ename: 'None Registered!'
+                                        })
+                                    }
+                                }
+                            }
+                        })
+                    }
+                } else {
+                    setNextEvent({
+                        ename: 'None Registered!'
+                    })
+                }
+            })
+    }
+
+    // set featured event and nextEvent on initial render
+    const initialRender = async () => {
+        if (!featuredEvent && !nextEvent) {
+            console.log('about to check events')
+            if (!props.events) {
+                console.log('UPDATING EVENTS RIGHT NOW')
+                await updateEvents()
```

Comment:
don't need to await, also 🔪 the console logs

Reply:
it breaks if i dont use await, but i'll remove the console.logs

Reply:
oh wait nvm,let me play around with it since i added the `if (props.events)` checks